src/pocketmine/item/Item.php

```diff
@@ -596,7 +596,13 @@ public function addEnchantment(Enchantment $ench){
 		}
 
 		if(!$found){
-			$tag->ench->{count($tag->ench) + 1} = new CompoundTag("", [
+			$count = 0;
+			foreach($tag->ench as $key => $value){
+				if(is_numeric($key)){
+					++$count;
+				}
+			}
+			$tag->ench->{$count + 1} = new CompoundTag("", [
```

Comment:
only one enchant could be added before, tested on latest commit without this commit.